File: front-end/src/app/models/order.ts
import {Cardorder} from './cardorder';

export class Order {
  id: number;
  tableNumber: number;
  totalPrice : number;
  orderItemDtos : Cardorder[];
  totalNumber : number;
  customerId : number;

  constructor(tableNumber: number, totalPrice: number,
  orderItemDtos:Cardorder[],totalNumber : number,
              customerId : number,id?: number) {

  }
}

File: resturant904/src/main/java/com/spring/resturant/model/Order.java
package com.spring.resturant.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "order_seq")
    @SequenceGenerator(
            name = "order_seq",
            sequenceName = "ORDER_SEQ",
            allocationSize = 1
    )
    private Long id;

    private Integer tableNumber;

    private String code;

    private Double totalPrice;

    private Integer totalNumber;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    private List<OrderItem> orderItems;

    private LocalDateTime atdate;

    @ManyToOne(fetch = FetchType.LAZY,
            cascade = CascadeType.ALL)
    @JoinColumn(name = "customer_id")
    private Customer customer;
}
